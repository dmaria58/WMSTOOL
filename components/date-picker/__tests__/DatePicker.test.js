import React from 'react';
import { mount } from 'enzyme';
import moment from 'moment';
import MockDate from 'mockdate';
import DatePicker from '..';
import focusTest from '../../../tests/shared/focusTest';

describe('DatePicker', () => {
  focusTest(DatePicker);

  it('prop locale should works', () => {
    const locale = {
      lang: {
        placeholder: 'Избери дата',
        rangePlaceholder: [
          'Начална дата',
          'Крайна дата',
        ],
        today: 'Днес',
        now: 'Сега',
        backToToday: 'Към днес',
        ok: 'Добре',
        clear: 'Изчистване',
        month: 'Месец',
        year: 'Година',
        timeSelect: 'Избор на час',
        dateSelect: 'Избор на дата',
        monthSelect: 'Избор на месец',
        yearSelect: 'Избор на година',
        decadeSelect: 'Десетилетие',
        previousMonth: 'Предишен месец (PageUp)',
        nextMonth: 'Следващ месец (PageDown)',
        previousYear: 'Последна година (Control + left)',
        nextYear: 'Следваща година (Control + right)',
        previousDecade: 'Предишно десетилетие',
        nextDecade: 'Следващо десетилетие',
        previousCentury: 'Последен век',
        nextCentury: 'Следващ век',
        yearFormat: 'YYYY',
        dateFormat: 'D M YYYY',
        dayFormat: 'D',
        dateTimeFormat: 'D M YYYY HH:mm:ss',
        monthBeforeYear: true,
      },
      timePickerLocale: {
        placeholder: 'Избор на час',
      },
    };
    MockDate.set(new Date('2016-11-22').getTime() + (new Date().getTimezoneOffset() * 60 * 1000));
    const birthday = moment('2000-01-01', 'YYYY-MM-DD');
    const wrapper = mount(
      <DatePicker open locale={locale} value={birthday} />
    );
    expect(wrapper.render()).toMatchSnapshot();
    MockDate.reset();
  });
});
