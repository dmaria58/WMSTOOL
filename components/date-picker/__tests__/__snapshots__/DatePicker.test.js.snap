// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker prop locale should works 1`] = `
<span
  class="ant-calendar-picker"
>
  <div
    style=""
  >
    <input
      class="ant-calendar-picker-input ant-input"
      placeholder="Избери дата"
      readonly=""
      value="2000-01-01"
    />
    <i
      class="anticon anticon-cross-circle ant-calendar-picker-clear"
    />
    <span
      class="ant-calendar-picker-icon"
    />
  </div>
  <div>
    <div
      class="ant-calendar-picker-container ant-calendar-picker-container-placement-bottomLeft slide-up-appear"
      style="left: -999px; top: -1002px;"
    >
      <div
        class="ant-calendar"
        tabindex="0"
      >
        <div
          class="ant-calendar-panel"
        >
          <div
            class="ant-calendar-input-wrap"
          >
            <div
              class="ant-calendar-date-input-wrap"
            >
              <input
                class="ant-calendar-input "
                placeholder="Избери дата"
                value="2000-01-01"
              />
            </div>
            <a
              class="ant-calendar-clear-btn"
              role="button"
              title="Изчистване"
            />
          </div>
          <div
            class="ant-calendar-date-panel"
          >
            <div
              class="ant-calendar-header"
            >
              <div
                style="position: relative;"
              >
                <a
                  class="ant-calendar-prev-year-btn"
                  role="button"
                  title="Последна година (Control + left)"
                />
                <a
                  class="ant-calendar-prev-month-btn"
                  role="button"
                  title="Предишен месец (PageUp)"
                />
                <span
                  class="ant-calendar-my-select"
                >
                  <a
                    class="ant-calendar-month-select"
                    role="button"
                    title="Избор на месец"
                  >
                    Jan
                  </a>
                  <a
                    class="ant-calendar-year-select"
                    role="button"
                    title="Избор на година"
                  >
                    2000
                  </a>
                </span>
                <a
                  class="ant-calendar-next-month-btn"
                  title="Следващ месец (PageDown)"
                />
                <a
                  class="ant-calendar-next-year-btn"
                  title="Следваща година (Control + right)"
                />
              </div>
            </div>
            <div
              class="ant-calendar-body"
            >
              <table
                cellspacing="0"
                class="ant-calendar-table"
                role="grid"
              >
                <thead>
                  <tr
                    role="row"
                  >
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Sun"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Su
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Mon"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Mo
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Tue"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Tu
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Wed"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        We
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Thu"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Th
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Fri"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Fr
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Sat"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Sa
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-calendar-tbody"
                >
                  <tr
                    class="ant-calendar-active-week"
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 26, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        26
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 27, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        27
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 28, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        28
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 29, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        29
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 30, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        30
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 31, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        31
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-selected-date ant-calendar-selected-day"
                      role="gridcell"
                      title="January 1, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="true"
                        class="ant-calendar-date"
                      >
                        1
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 2, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        2
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 3, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        3
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 4, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        4
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 5, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        5
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 6, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        6
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 7, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        7
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 8, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        8
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 9, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        9
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 10, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        10
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 11, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        11
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 12, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        12
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 13, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        13
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 14, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        14
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 15, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        15
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 16, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        16
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 17, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        17
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 18, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        18
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 19, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        19
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 20, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        20
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 21, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        21
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 22, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        22
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 23, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        23
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 24, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        24
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 25, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        25
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 26, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        26
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 27, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        27
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 28, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        28
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 29, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        29
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 30, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        30
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 31, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        31
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 1, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        1
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 2, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        2
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 3, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        3
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 4, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        4
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 5, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        5
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="ant-calendar-footer"
            >
              <span
                class="ant-calendar-footer-btn"
              >
                <a
                  class="ant-calendar-today-btn "
                  role="button"
                  title="November 22, 2016"
                >
                  Днес
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
